Integrate changes into dev branch.

damienmarchal opened the pull request:
CHANGELOG:
- CameraView now use the BackgroundSetting 
- Adds an ObjectRenderer to dispatch the rendering of sofa object the way runSofa2 wants it
- Renames ControlledSofaView into CameraView
- Renames SofaViewer into EditView
- Fix implementation of getComponentsbyType that was not working with parent class. 
- Change RGBAColor so they are reported as array in AbstractTypeInfo
- FEATURE: Go to source code emission loc
- FEATURE: Go to scene file location
- Filter error messages by selected component

damienmarchal committed:
CHANGE the implementation of getComponentsByType

It was not working when using a parent type.

damienmarchal committed:
RENAME ControlledSofaView to CameraView and make it display the BaseCamera instead of InteractiveCamera

Most of the time, interaction are "application" specific.
So we should implement the movement of the camera and the connection to
user input as external code (eg: PythonScriptController).

So it better the widget to display BaseCamera so that anyone can take profit of this
view to render the "in-scenes" cameras.

damienmarchal committed:
REMOVE ControlledSofaViewer.*

damienmarchal committed:
ADD CameraView and EditView

The two are different implementation of the 3D view.
CameraView is for in scene camera rendering, all the events should be passed
to the Sofa pipeline. We can only have one CameraView (otherwise it will be a
giant mess to forward the user inputs through the scene graph).

EditView is the 'editors' view where we can interact with editor's tools.

damienmarchal committed:
Move the 3Dviews into a dedicated directory.

damienmarchal committed:
SofaCamera.cpp small cleaning.

damienmarchal committed:
Change "Inherit" to "Parent"

damienmarchal committed:
SofaScene, SofaViewer,CameraView: Refactor to factor at the right place the rendering code.

To me te rendering code shouldn't be in SofaScene.

We should have:
MainEditor (which contains elements shared by all views)
EditView   (which render from one view point in editor mode)
CameraView (which render from scene mode)
SofaScene  (which only work on the scene itself)

damienmarchal committed:
ADD ObjectRenderer to put all editor's rendering operation.

damienmarchal committed:
REMOVE from CameraView.qml the color&background properties.

damienmarchal committed:
ADD in CameraView.* the rendering of BackgroundColor&Image from the camera.

damienmarchal committed:
UPDATE the RGBAColor widget to handle the change of typeinfo.

damienmarchal committed:
REFACTOR 3D rendering organization in SofaView/SofaScene/SofaCamera.

fredroy commented on applications-dev/plugins/SofaQtQuickGUI/SofaViewer.cpp:
```diff
@@ -505,7 +505,7 @@ QPair<QVector3D, QVector3D> SofaViewer::boundingBox() const
 {
 	QVector3D min, max;
-	mySofaScene->computeBoundingBox(min, max, myRoots); /*attention, ne prend en compte que le dernier �l�ment mis dans la liste ?*/
+	mySofaScene->computeBoundingBox(min, max, myRoots); /*attention, ne prend en compte que le dernier �l�ment mis dans la liste ?*/
```
I guess it should translated in english....

damienmarchal replied:
Now I understand why my editor was complaining all the time. 

If only 
*--coding-utf-8--  
was supported in c++

fredroy commented on applications-dev/plugins/SofaQtQuickGUI/windows/CameraView.cpp:
```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2018, CNRS,INRIQ
```
INRIQ ?

damienmarchal committed:
ADD MainStyle to centralize color and style stuff.

damienmarchal committed:
ADD Support for debugging.

damienmarchal committed:
ADD SofaLink (to manipulate link in the qml part)

damienmarchal committed:
CHANGE CMakeLists to add debugging support & new files.

damienmarchal committed:
ADD function to manipulate link in SofaScene.

damienmarchal committed:
ADD MainStyle.qml & use it SofaApplication to make a 'style' property

damienmarchal committed:
ADD MainStyle in qrc.

damienmarchal committed:
USE MainStyle in SofaViewer

damienmarchal committed:
ADD MainSTyle in qmldir

damienmarchal committed:
ADD a preliminary openInEditor() function in SofaApplication.

This function can be used to open any file at a given location.
Currently this function only works for qtcreator. Other tools
may be supported (probably using a pattern filling approach).

damienmarchal committed:
BUGIFX set nullptr as the default value for pointer in SofaCamera.

Otherwise testing against nullptr may fails;

damienmarchal committed:
BUFIX in CameraView due to nullptr access.

damienmarchal committed:
CHANGE Hide Messages, Show "Infos" and add debug button in the inspector.

damienmarchal committed:
RENAME SofaWidget into SofaView

'Views' are integrating a lot of functionnality into a top level panel
that can be displayed as is to the user eg: Inspector, CameraView, EditView,
SceneGraphView. A view is connected to the workflow of user and how they set they
their workspace.

Widget is a more simple element, a single widget can be integrated into multiple views.
Eg of widget: Header, SearchBar,

damienmarchal committed:
[SofaQtQuick] REFACTOR by putting the SearchBar out of SofaSceneGraphView.

damienmarchal committed:
ADD SofaNodeMenu, SofaObjectMenu, SofaWindowDataListView from SofaSceneGraphiView.

damienmarchal committed:
CHANGE Minor schema in SofaDataListView

damienmarchal committed:
FIX in qmldir

damienmarchal committed:
FILEMOVE *.cpp *.h *.inl to src/SofaQtQuickGUI/

Follow the SofaNG standard for encapsulating packages.

damienmarchal committed:
MOVE FILE WIP

damienmarchal committed:
ADD test stub for SofaScene

damienmarchal committed:
ADD A signal to visually show which object is selected.

damienmarchal committed:
ADD in SofaComponent method to handle file location.

This is usefull to jump to source code & scene file at correct location.
(Warning the feature currently work only with qtcreator)

damienmarchal committed:
ADD two new windows for Message & Help.

damienmarchal committed:
ADD the new windows to qmldir & qrc.

damienmarchal committed:
ADD new windows to qmldir (wip)

damienmarchal committed:
REMOVE 'onClosing' in SofaWindowDatalistView

I have an error on this line in qt 5.11.

damienmarchal committed:
ADD "go to" feature to SofaSceneGraphView

*) It is now possible from the message view to go to select the emitter's message
to select it in the scenegraph view

*) In the popup menu for Node, it is now possible to jump to source code location of the implementation
and to the location in the scene file where it is used.

damienmarchal committed:
FIX "Go to" functions that were not working properly when the object does not have the adequate code location.

damienmarchal committed:
FIX handling of BaseNode in SofaComponent.cpp

damienmarchal committed:
ADD Condition to restrict the display only on selected entries;

damienmarchal committed:
REMOVE the number of selected message because it cause a loop in QML.

If anyone knows how to restore this behavior I would be very happy.

damienmarchal committed:
FIX alignement in SofaApplication.qml

damienmarchal committed:
ADD the SofaMessageList as a property of SofaApplication.

damienmarchal committed:
ADD clearing of SofaMessageList when the scene is reloaded.

damienmarchal committed:
ADD a dedicated windows to display a message for a given component through pop-up menu

damienmarchal committed:
CLEAN SofaNodeMenu & SofaWindowComponentMessage

damienmarchal committed:
Add menu entries in DefaultMenuBar.qml (disabled with a TODO)

damienmarchal committed:
Fix Bug on Console.cpp that was crashing runSofa2 when it was requested to signal a component.

damienmarchal committed:
REMOVE not needed code in SofaInspectorDataModel.cpp

damienmarchal committed:
FIX Inspector.qml bug

The header is now correct and the links are now displayed the proper way.

damienmarchal committed:
Improve the message.qml view

damienmarchal committed:
FIX SofaScene::addNodeTo() so that it returns a SofaComponent* instead of bool

damienmarchal committed:
ADD ugly linkArrow icon. Should be redone by a designer.

damienmarchal committed:
ADD multilink autojump & signalling to SofaSceneGraphView

All the icons in the rows are now taking the same spacing to avoid having
zig-zagging hierarhcy.

The rendering is changed for multi-parent node to allow to differentiate them more easily.
Double clikcing on a multiparent node "auto jump" the selected object to the
implementation.

Implement a way to visually signal (by blinking) that the selection
is changing.




 the first instance is rendered differently, by double clikcing on
a multi-parent node it is now possible to "autojump" to its "details".

damienmarchal committed:
SofaSceneGraphView.qml restore the "disabled" sign.

damienmarchal committed:
ADD to SofaComponent a function to get its parent.

damienmarchal committed:
ADD in SofaScene::addNodeTo autoincrement the name XXXX0 so that there is no duplicated name in the graph.

damienmarchal committed:
ADD Add to SofaNodeMenu  "Add sibling"

damienmarchal committed:
[SofaQtQuickGUI] Move Manipulators file into a Manipulator directory (1/2)

damienmarchal committed:
[SofaQtQuickGUI] Update #include to "Manipulators/Manipulators*.h" (2/2)

damienmarchal committed:
[SofaQtQuickGUI] ADD some testing infrastucture;

damienmarchal committed:
[SofaQtQuickGUI] Remove commented code;

damienmarchal committed:
[SofaQtQuickGUI] Add SofaFactory to expose the factory into QML.

damienmarchal committed:
[SofaQtQuickGUI] Work on SofaSceneListModel.*

- Adds override in methode declaration.
- work on addNode function

damienmarchal committed:
[SofaQtQuickGUI] ADD new method to create a component by string.

damienmarchal committed:
[SofaQtQuickGUI] QML new code & change in SofaNodeMenu, and PopupWindowCreateComponent

damienmarchal committed:
[SofaQtQuickGUI] Update SofaScene.cpp so it include "Rendering/ObjectRenderer.h"

damienmarchal committed:
Merge remote-tracking branch 'upstream/dev' into dev
